### ISSUE TITLE
[BUG]: x of facial area left and right eye seems to be swapped since version 0.0.90

### ISSUE THREAD

**Issue by tomwieden** [serengil/deepface#1217]
### Before You Report a Bug, Please Confirm You Have Done The Following...

- [X] I have updated to the latest version of the packages.
- [X] I have searched for both [existing issues](https://github.com/serengil/deepface/issues) and [closed issues](https://github.com/serengil/deepface/issues?q=is%3Aissue+is%3Aclosed) and found none that matched my issue.

### DeepFace's version

v0.0.90

### Python version

Python 3.10.14

### Operating System

Debian testing

### Dependencies

sh-5.1$ pip freeze
absl-py==2.1.0
astunparse==1.6.3
attrs==22.2.0
beautifulsoup4==4.12.3
blinker==1.7.0
build==0.10.0
CacheControl==0.12.11
cachetools==5.3.3
certifi==2022.12.7
cffi==1.15.1
charset-normalizer==3.1.0
cleo==2.0.1
click==8.1.7
crashtest==0.4.1
cryptography==39.0.2
Cython @ file:///buildstream-build
deepface==0.0.90
distlib==0.3.6
dulwich==0.21.3
filelock==3.9.0
fire==0.6.0
Flask==3.0.3
flatbuffers==24.3.25
gast==0.5.4
gdown==5.1.0
google-auth==2.29.0
google-auth-oauthlib==1.2.0
google-pasta==0.2.0
grpcio==1.62.2
gunicorn==22.0.0
h5py==3.11.0
html5lib==1.1
idna==3.4
importlib-metadata==6.0.0
installer==0.6.0
itsdangerous==2.2.0
jaraco.classes==3.2.3
jeepney==0.8.0
Jinja2==3.1.3
jsonschema==4.17.3
keras==2.15.0
keyring==23.13.1
libclang==18.1.1
lockfile==0.12.2
Mako @ file:///buildstream-build
Markdown @ file:///buildstream-build
markdown-it-py==3.0.0
MarkupSafe @ file:///buildstream-build
mdurl==0.1.2
meson==0.63.3
ml-dtypes==0.2.0
more-itertools==9.1.0
msgpack==1.0.5
mtcnn==0.1.1
namex==0.0.8
numpy==1.26.4
oauthlib==3.2.2
opencv-python==4.9.0.80
opt-einsum==3.3.0
optree==0.11.0
packaging==23.0
pandas==2.2.2
pexpect==4.8.0
pillow==10.3.0
pipenv==2023.2.18
pkginfo==1.9.6
platformdirs==2.6.2
poetry==1.4.0
poetry-core==1.5.1
poetry-plugin-export==1.3.0
protobuf==4.25.3
ptyprocess==0.7.0
pyasn1==0.6.0
pyasn1_modules==0.4.0
pycparser==2.21
Pygments @ file:///buildstream-build
pyproject_hooks==1.0.0
pyrsistent==0.19.3
PySocks==1.7.1
python-dateutil==2.9.0.post0
pytz==2024.1
rapidfuzz==2.13.7
requests==2.28.2
requests-oauthlib==2.0.0
requests-toolbelt==0.10.1
retina-face==0.0.17
rich==13.7.1
rsa==4.9
SecretStorage==3.3.3
shellingham==1.5.0.post1
six @ file:///buildstream-build
soupsieve==2.5
tensorboard==2.15.2
tensorboard-data-server==0.7.2
tensorflow==2.15.0
tensorflow-estimator==2.15.0
tensorflow-io-gcs-filesystem==0.36.0
termcolor==2.4.0
tf-keras==2.15.0
tomli==2.0.1
tomlkit==0.11.6
tqdm==4.66.2
trove-classifiers==2023.3.9
typing_extensions==4.11.0
tzdata==2024.1
urllib3==1.26.15
virtualenv==20.20.0
virtualenv-clone==0.5.7
webencodings==0.5.1
Werkzeug==3.0.2
wrapt==1.14.1
zipp==3.15.0

### Reproducible example

```Python
resp_objs = DeepFace.represent(
                    img_path=file,
                    model_name=Facenet512,
                    enforce_detection=False,
                    detector_backend=retinaface,
                    align=True,
                    expand_percentage=0,
                    normalization="base",
                )

for resp_obj in resp_objs:
    x_left_eye = resp_obj['facial_area']['left_eye'][0]
    x_right_eye = resp_obj['facial_area']['right_eye'][0]
    w_eye_distance = x_right_eye - x_left_eye
```

### Expected Result

w_eye_distance should be a positive integer for a person that is looking into the camera

### What happened instead?

w_eye_distance is now negative where it was positive in version prior 0.0.90

### Additional Info

Does anybody experience the same?

Note on tf-keras version...

After installing deepace on a fresh install of Debian testing...

     pip install deepface

I had to downgrade keras

    pip install --upgrade tf-keras==2.15 tensorflow==2.15

because when the code that is using deepface was executed it complained about a wrong tf-keras version. That's why you see tf-keras==2.15.0 in the output of pip freeze.

**Comment by serengil** [comment]
left eye is now the left eye of the person itself, not from the observer.

**Comment by serengil** [comment]
So, this is not a bug, an expected behaviour.

**Comment by tomwieden** [comment]
Got it thanks, my code uses the abs(w_eye_distance) now.

**Comment by tomwieden** [comment]
Should I file a bug on the tf-keras version that must be downgraded?

**Comment by serengil** [comment]
it is not a bug. if you use tf 2.16, installing tf-keras is expected.

**Comment by tomwieden** [comment]
Proposal/question...

Is it possible that

    pip install deepface

is using a tf version < 16 by default?

The only thing I install on a fresh Debian is

```
su -c 'apt update && apt install default-jre python3-pip python3.11-venv cmake libimage-exiftool-perl darktable'
python3 -m venv ~/.ddc/deepface && source ~/.ddc/deepface/bin/activate
pip install deepface mediapipe dlib facenet-pytorch ultralytics

pip install --upgrade tf-keras==2.15 tensorflow==2.15

pip show ultralytics
pip show facenet-pytorch
pip show mediapipe
pip show dlib
pip show deepface

deactivate
```

Everything works out of the box except tf-keras (this is a little wonder for me because of all the dependencies).

This will be my last comment on this. Promised ;-) !

**Comment by serengil** [comment]
unfortunately, requirements cannot be defined conditionally e.g. https://github.com/serengil/deepface/blob/master/requirements.txt

**Comment by tomwieden** [comment]
Got it. 

Thanks for your very quick replies!!
